tinygrad/ops.py

```diff
@@ -238,54 +239,45 @@ def backward(ctx, grad_output):
 
 # ************* pooling ops *************
 
-def stack_for_pool(x, py, px):
-  my, mx = (x.shape[2]//py)*py, (x.shape[3]//px)*px
-  stack = []
-  xup = x[:, :, :my, :mx]
-  for Y in range(py):
-    for X in range(px):
-      stack.append(xup[:, :, Y::py, X::px][None])
-  return np.concatenate(stack, axis=0)
-
-def unstack_for_pool(fxn, s, py, px):
+def pool(x, py, px, fxn=None):
```

Comment:
This really shouldn't be one function. The codepath is totally different depending on fxn.

The goal is not less lines. The goal is readable code first and foremost, then less lines. This is hard to read.

Reply:
alrite, will delete pr